src/main/java/sx/blah/discord/api/internal/DiscordVoiceWS.java

```diff
@@ -257,11 +257,13 @@ public void run() {
 	 * Sets up the thread for receiving audio packets from the Discord UDP Socket.
 	 */
 	private void setupReceiveThread() {
-		Runnable receiveThread = () -> {
+		(new Thread((Runnable) () -> {
+                    while(true)
 			if (isConnected.get()) {
 				DatagramPacket receivedPacket = new DatagramPacket(new byte[1920], 1920);
 				try {
 					udpSocket.receive(receivedPacket); // This blocks the thread until a packet is received.
```

Comment:
If you're planning to use this to block until the packet is received, then you should set SoTimeout on the udpSocket and catch SocketTimeoutException and ignore it so that you don't block the thread forever. If you block forever without a timeout you won't be able to interrupt the thread to shut it down.